src/picongpu/include/plugins/hdf5/WriteSpecies.hpp

```diff
@@ -346,13 +350,14 @@ struct WriteSpecies
          *         global domain offsets (slides), etc.
          * extent: size of this particle patch, upper bound is excluded
          */
+        const PMacc::Selection<simDim>& globalDomain = Environment<simDim>::get().SubGrid().getGlobalDomain();
```

Comment:
Note: If we use domain offsets those are not coming from the `param`  object than we can introduce side effects.
Currently we get the all domain definitions from `params`